### PR TITLE
Add table_aws_eks_cluster. Closes #122

### LINKED ISSUE #122: Add table_aws_eks_cluster

### PR DESCRIPTION
**Pull request by rajeshbal65**
SETUP: tests/aws_eks_cluster []

PRETEST: tests/aws_eks_cluster

TEST: tests/aws_eks_cluster
Running terraform
aws_iam_role.named_test_resource: Creating...
aws_vpc.named_test_resource: Creating...
aws_iam_role.named_test_resource: Creation complete after 7s [id=turbottest6230]
aws_iam_role_policy_attachment.named_test_resource: Creating...
aws_iam_role_policy_attachment.named_test_resource2: Creating...
aws_vpc.named_test_resource: Still creating... [10s elapsed]
aws_iam_role_policy_attachment.named_test_resource: Creation complete after 4s [id=turbottest6230-20210316105121589200000001]
aws_iam_role_policy_attachment.named_test_resource2: Creation complete after 4s [id=turbottest6230-20210316105121653100000002]
aws_vpc.named_test_resource: Still creating... [20s elapsed]
aws_vpc.named_test_resource: Creation complete after 24s [id=vpc-0a71f3447a0c8d50e]
aws_subnet.named_test_resource1: Creating...
aws_subnet.named_test_resource2: Creating...
aws_subnet.named_test_resource2: Creation complete after 7s [id=subnet-0bec20738638a547f]
aws_subnet.named_test_resource1: Creation complete after 7s [id=subnet-0300cded80d8d77b2]
aws_eks_cluster.named_test_resource: Creating...
aws_eks_cluster.named_test_resource: Still creating... [10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [10m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [10m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [10m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [10m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [10m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [10m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [11m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [11m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [11m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [11m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [11m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [11m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [12m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [12m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [12m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [12m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [12m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [12m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [13m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [13m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [13m20s elapsed]
aws_eks_cluster.named_test_resource: Creation complete after 13m21s [id=turbottest6230]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals

Apply complete! Resources: 7 added, 0 changed, 0 destroyed.

Outputs:

account_id = "352685002396"
endpoint = "https://D25E242F3322C22FEC865FEC79A7F810.gr7.us-east-1.eks.amazonaws.com"
identity = tolist([
  {
    "oidc" = tolist([
      {
        "issuer" = "https://oidc.eks.us-east-1.amazonaws.com/id/D25E242F3322C22FEC865FEC79A7F810"
      },
    ])
  },
])
kubeconfig-certificate-authority-data = "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUN5RENDQWJDZ0F3SUJBZ0lCQURBTkJna3Foa2lHOXcwQkFRc0ZBREFWTVJNd0VRWURWUVFERXdwcmRXSmwKY201bGRHVnpNQjRYRFRJeE1ETXhOakV4TURBek1sb1hEVE14TURNeE5ERXhNREF6TWxvd0ZURVRNQkVHQTFVRQpBeE1LYTNWaVpYSnVaWFJsY3pDQ0FTSXdEUVlKS29aSWh2Y05BUUVCQlFBRGdnRVBBRENDQVFvQ2dnRUJBTHRlCkdjSDg2S09FQ1BNWXppRUxhejAza1pQUzZTQWF3K3hSbjBNZ0hHK1EwT0FKbFZSejJ4VGpJOUgzZFdNb241Q04KWlR0eWZDcDdnVUdIZ0ZkWmVnRDZ4VUhNS1hncDRhd29pdC9QSGlGc05kaWd5SnMwQWhFaEJQYUJya3E3RVg0Wgp6VDh6KzRTdjYwUks1Y29nKzA2d2x6cUQxazVZdTRTaitXbzUvSCtoYjRtblk5andVOWcybndCVTN2QlFZdVJUCnJnOUhGUDI2bVM3bk80alBBQ0MzanVIR1UvMXNId0Q4cjJIajBFRm9aZTJ4aFNJeXdYM3h0S0h4czdINXRnQUsKZkNSTndTajFQSEhEL2s2KzZqY08rZm0xa1pZOVhtbUtQWFVDSXJGUG5pbEVXNjVqampRVGpNVktSaUVRaGdCUgord2JNam43YjBtM3JpNG1zK0NNQ0F3RUFBYU1qTUNFd0RnWURWUjBQQVFIL0JBUURBZ0trTUE4R0ExVWRFd0VCCi93UUZNQU1CQWY4d0RRWUpLb1pJaHZjTkFRRUxCUUFEZ2dFQkFEUnl6VXo4ajd3RW14SkdPNk8vaWoxUHkrWnQKby9MOXBmR3pqSXlpOFJZYzZGZUd1UnJnZkdnUTlnVWFMZ0xsZ04zd3RyQUhYRmVqS0k4dDlIMnViRXVRVzlBUQptRDZyQ2dFVlVRTDlhTTNBZDJHOVBWSlJ0QmJmTkw1UlNWL0hKMmtRVXk0TE1XcTNsTUdKekU2UXB3UVNJMy82ClZta3FySGNXQUhwcGFhY2Z0MnpxeWlxT05uc3h3bVIvSWxOUUJuak1kWE9xWnRveUdVNlhLbzZ1ellFWXdRK1kKOEdqemtGTXhZV1ZBcDlNQWYweTZaZ1VPZ2JvYW9vQ0lRbFZSdzY1QjAvd3JKSjBadlZYTWZVV0dMczd2b1lVMAp4SzIzcEVVZWMyL2NZYmc3emJBc2t6VDkvemhsbGZEZ0VwYXh2Y2NSVWpJc2Fsb3Vqc1FuaWpXMlZuMD0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo="
platform_version = "eks.3"
resource_aka = "arn:aws:eks:us-east-1:352685002396:cluster/turbottest6230"
resource_id = "turbottest6230"
resource_name = "turbottest6230"
role_arn = "arn:aws:iam::352685002396:role/turbottest6230"
status = "ACTIVE"
version = "1.18"
vpc_config = tolist([
  {
    "cluster_security_group_id" = "sg-003dc6d65b131bf55"
    "endpoint_private_access" = false
    "endpoint_public_access" = true
    "public_access_cidrs" = toset([
      "0.0.0.0/0",
    ])
    "security_group_ids" = toset(null) /* of string */
    "subnet_ids" = toset([
      "subnet-0300cded80d8d77b2",
      "subnet-0bec20738638a547f",
    ])
    "vpc_id" = "vpc-0a71f3447a0c8d50e"
  },
])

Running SQL query: test-get-query.sql
[
  {
    "arn": "arn:aws:eks:us-east-1:352685002396:cluster/turbottest6230",
    "endpoint": "https://D25E242F3322C22FEC865FEC79A7F810.gr7.us-east-1.eks.amazonaws.com",
    "name": "turbottest6230",
    "platform_version": "eks.3",
    "role_arn": "arn:aws:iam::352685002396:role/turbottest6230",
    "status": "ACTIVE",
    "version": "1.18"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:eks:us-east-1:352685002396:cluster/turbottest6230",
    "endpoint": "https://D25E242F3322C22FEC865FEC79A7F810.gr7.us-east-1.eks.amazonaws.com",
    "name": "turbottest6230",
    "status": "ACTIVE"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "352685002396",
    "akas": [
      "arn:aws:eks:us-east-1:352685002396:cluster/turbottest6230"
    ],
    "name": "turbottest6230",
    "tags": {
      "Name": "turbottest6230"
    },
    "title": "turbottest6230"
  }
]
✔ PASSED

POSTTEST: tests/aws_eks_cluster

TEARDOWN: tests/aws_eks_cluster

SUMMARY:

1/1 passed.

### CODE REVIEW

File: aws-test/tests/aws_eks_cluster/test-get-query.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+select name, arn, endpoint, role_arn, version, resources_vpc_config, identity, status, certificate_authority -> 'Data'as certificate_authority_data
+from aws.aws_eks_cluster
+where arn = '{{ output.resource_aka.value }}'
```
Comment (khushboo9024) [inline aws-test/tests/aws_eks_cluster/test-get-query.sql:3]: Use `name` to check get query.

Comment (khushboo9024) [inline aws-test/tests/aws_eks_cluster/test-get-query.sql:3]:
```suggestion
where name = '{{ resourceName }}';
```

File: aws-test/tests/aws_eks_cluster/test-list-query.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+select name, arn, endpoint, status, role_arn
+from aws.aws_eks_cluster
+where arn = '{{ output.resource_aka.value }}'
```
Comment (khushboo9024) [inline aws-test/tests/aws_eks_cluster/test-list-query.sql:3]:
```suggestion
where arn = '{{ output.resource_aka.value }}';
```

File: aws-test/tests/aws_eks_cluster/test-not-found-query.sql [hunk 3]
```diff
@@ -0,0 +1,3 @@
+select name, title, tags, akas, account_id
+from aws.aws_eks_cluster
+where name = 'dummy-{{ resourceName }}'
```
Comment (khushboo9024) [inline aws-test/tests/aws_eks_cluster/test-not-found-query.sql:3]:
```suggestion
where name = 'dummy-{{ resourceName }}';
```

File: aws-test/tests/aws_eks_cluster/test-turbot-query.sql [hunk 4]
```diff
@@ -0,0 +1,3 @@
+select name, title, tags, akas, account_id
+from aws.aws_eks_cluster
+where name = 'dummy-{{ resourceName }}'
```
Comment (khushboo9024) [inline aws-test/tests/aws_eks_cluster/test-turbot-query.sql:3]:
```suggestion
where name = 'dummy-{{ resourceName }}';
```

File: aws/table_aws_eks_cluster.go [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listEksClusters(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_eks_cluster.go:134]:
```suggestion
```

File: aws/table_aws_eks_cluster.go [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+	}
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getEksCluster(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getEksCluster")
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_eks_cluster.go:162]:
```suggestion
```

File: docs/tables/aws_eks_cluster.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+```sql
+select
+  name,
+  arn,
+  endpoint,
+  identity,
+  status
+from
+  aws_eks_cluster;;
+```
```
Comment (khushboo9024) [inline docs/tables/aws_eks_cluster.md:18]: Please add few more queries

File: aws/table_aws_eks_cluster.go [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("tags"),
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "tags",
+				Description: "A list of tags assigned to the table",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEksCluster,
```
Comment (khushboo9024) [inline aws/table_aws_eks_cluster.go:108]: We should try to keep all the JSON fields at end of the table, please change accordingly

File: docs/tables/aws_eks_cluster.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+```sql
+select
+  name,
+  arn,
+  endpoint,
+  identity,
+  status
+from
+  aws_eks_cluster;;
```
Comment (khushboo9024) [inline docs/tables/aws_eks_cluster.md:17]:
```suggestion
  aws_eks_cluster;
```

File: aws/table_aws_eks_cluster.go [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+				Name:        "platform_version",
+				Description: "The platform version of your Amazon EKS cluster.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "tags_src",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("tags"),
```
Comment (khushboo9024) [inline aws/table_aws_eks_cluster.go:101]: we are getting the tags in TagsMap format the why we are adding `tags_src` column.

File: aws/table_aws_eks_cluster.go [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+import (
+	"context"
+
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+
+	"github.com/aws/aws-sdk-go/service/eks"
+)
+
```
Comment (khushboo9024) [inline aws/table_aws_eks_cluster.go:12]:
```suggestion
//// TABLE DEFINITION
```

File: aws-test/tests/aws_eks_cluster/variables.tf [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+  provider = aws.alternate
+}
+
+data "null_data_source" "resource" {
+  inputs = {
+    scope = "arn:${data.aws_partition.current.partition}:::${data.aws_caller_identity.current.account_id}"
+  }
+}
+
+# Create AWS > EC2 > Transit Gateway
```
Comment (khushboo9024) [inline aws-test/tests/aws_eks_cluster/variables.tf:50]:
```suggestion
# Create AWS > EKS > Cluster
```

File: aws/table_aws_eks_cluster.go [hunk 13]
```diff
@@ -0,0 +1,192 @@
...
+			},
+			{
+				Name:        "arn",
+				Description: "The Amazon Resource Name (ARN) of the cluster.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "created_at",
+				Description: "The Time when the cluster was created.",
```
Comment (khushboo9024) [inline aws/table_aws_eks_cluster.go:39]:
```suggestion
				Description: "The Unix epoch timestamp in seconds for when the cluster was created.",
```

File: aws/table_aws_eks_cluster.go [hunk 14]
```diff
@@ -0,0 +1,192 @@
...
+				Description: "The certificate-authority-data for the cluster.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "platform_version",
+				Description: "The platform version of your Amazon EKS cluster.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksCluster,
+			},
```
Comment (khushboo9024) [inline aws/table_aws_eks_cluster.go:96]: There are few more properties like `KubernetesNetworkConfig` and `EncryptionConfig` which are not added.

**Review by khushboo9024 — COMMENTED** [review]
Please add integration test logs also.

File: aws/table_aws_eks_cluster.go [hunk 15]
```diff
@@ -0,0 +1,192 @@
...
+	params := &eks.DescribeClusterInput{
+		Name: &clusterName,
+	}
+
+	op, err := svc.DescribeCluster(params)
+	if err != nil {
+		return nil, err
+	}
+
+	return op.Cluster, nil
```
Comment (khushboo9024) [inline aws/table_aws_eks_cluster.go:191]:
```suggestion
	if op.Cluster != nil {
		return op.Cluster, nil
	}
return nil, nil
```

File: aws/table_aws_eks_cluster.go [hunk 16]
```diff
@@ -0,0 +1,200 @@
...
+			{
+				Name:        "arn",
+				Description: "The Amazon Resource Name (ARN) of the cluster.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "created_at",
+				Description: "The Unix epoch timestamp in seconds for when the cluster was created.",
+				Type:        proto.ColumnType_STRING,
```
Comment (Subhajit97) [inline aws/table_aws_eks_cluster.go:42]:
```suggestion
				Type:        proto.ColumnType_TIMESTAMP,
```

File: aws/table_aws_eks_cluster.go [hunk 17]
```diff
@@ -0,0 +1,200 @@
...
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "platform_version",
+				Description: "The platform version of your Amazon EKS cluster.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "tags",
```
Comment (Subhajit97) [inline aws/table_aws_eks_cluster.go:112]: Though the values are same, we need to define both `tags_src` and `tags`.

File: aws/table_aws_eks_cluster.go [hunk 18]
```diff
@@ -0,0 +1,200 @@
...
+			},
+
+			// Standard columns for all tables
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+				Hydrate:     getEksCluster,
+			},
```
Comment (Subhajit97) [inline aws/table_aws_eks_cluster.go:125]: Add `tags` column

File: aws/table_aws_eks_cluster.go [hunk 19]
```diff
@@ -0,0 +1,200 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := svc.ListClusters(&eks.ListClustersInput{})
+	for _, cluster := range resp.Clusters {
+		d.StreamListItem(ctx, &eks.Cluster{
+			Name: cluster,
+		})
+	}
```
Comment (Subhajit97) [inline aws/table_aws_eks_cluster.go:158]: Handle pagination
https://github.com/aws/aws-sdk-go/blob/ae393afe895c4d711a69c7d4a4570a48b1c86179/service/eks/api.go#L2210

File: aws/table_aws_eks_cluster.go [hunk 20]
```diff
@@ -0,0 +1,200 @@
...
+
+	op, err := svc.DescribeCluster(params)
+	if err != nil {
+		return nil, err
+	}
+
+	if op.Cluster != nil {
+		return op.Cluster, nil
+	}
```
Comment (Subhajit97) [inline aws/table_aws_eks_cluster.go:197]: Do we need this check?
Doesn't the API return any `Not Found` error?

File: docs/tables/aws_eks_cluster.md [hunk 21]
```diff
@@ -0,0 +1,72 @@
...
+  arn,
+  endpoint,
+  identity,
+  status
+from
+  aws_eks_cluster;
+```
+
+
+### List the Vpc configuration used by cluster
```
Comment (Subhajit97) [inline docs/tables/aws_eks_cluster.md:21]:
```suggestion
### List the VPC configuration used by cluster
```

File: aws-test/tests/aws_eks_cluster/variables.tf [hunk 22]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+resource "aws_vpc" "named_test_resource" {
+  cidr_block = "172.31.0.0/16"
+}
+
+resource "aws_subnet" "named_test_resource1" {
+  vpc_id     = aws_vpc.named_test_resource.id
+  cidr_block = "172.31.0.0/20"
+  availability_zone = "us-east-1b"
```
Comment (Subhajit97) [inline aws-test/tests/aws_eks_cluster/variables.tf:109]:
```suggestion
  availability_zone = "${var.aws_region}b"
```

File: aws-test/tests/aws_eks_cluster/variables.tf [hunk 23]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+resource "aws_subnet" "named_test_resource2" {
+  vpc_id     = aws_vpc.named_test_resource.id
+  cidr_block = "172.31.32.0/20"
+  availability_zone = "us-east-1d"
```
Comment (Subhajit97) [inline aws-test/tests/aws_eks_cluster/variables.tf:115]:
```suggestion
  availability_zone = "${var.aws_region}d"
```

File: docs/tables/aws_eks_cluster.md [hunk 24]
```diff
@@ -0,0 +1,72 @@
...
+```
+
+
+### List of log Types that are not enabled for logging
+
+```sql
+select
+  name,
+  i -> 'Enabled' as enabled,
+  i -> 'Types' as types
```
Comment (Subhajit97) [inline docs/tables/aws_eks_cluster.md:44]:
```suggestion
  i ->> 'Enabled' as enabled,
  i ->> 'Types' as types
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
Make suggested changes, and also add the output of the queries added in docs

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**